File: javascript/Mod4/4.03/test/LoginTest.js
// filename: test/LoginTest.js
//const { Builder } = require('selenium-webdriver') 
// const path = require('path')
const assert = require('assert')
const LoginPage = require('../pages/LoginPage')
require('./spec_helper')


//describe is a method from Mocha
describe('Login', function() { 
    //this.timeout(30000)
    //let driver 
    let login

    beforeEach(async function() { 
    //driver = await new Builder().forBrowser('firefox').build()
    login = new LoginPage(this.driver)
    await login.load()
    })

    // afterEach(async function() { 
    //     await driver.quit()
    // })

    it('with valid credentials', async function() {
    await login.authenticate('tomsmith', 'SuperSecretPassword!')
    assert(await login.successMessagePresent(), 'Success message not displayed')
  })
  
  it('not invalid credentials', async function() {
    await login.authenticate('tomsmith', 'SuperSecretPassword!')
    assert(!(await login.failureMessagePresent()), 'Failure message displayed')
  })

})

File: python/Mod3/3.05/tests/login_test.py
# filename: tests/login_test.py
import pytest
import os
from selenium import webdriver
from pages.login_page import LoginPage

@pytest.fixture
def login(request):
    _chromedriver = os.path.join(os.getcwd(), 'vendor', 'chromedriver')
    if os.path.isfile(_chromedriver):
        driver_ = webdriver.Chrome(_chromedriver)

    else:
        driver_ = webdriver.Chrome()

    loginPage = LoginPage(driver_)
    def quit():
        driver_.quit()

    request.addfinalizer(quit)
    return loginPage

def test_valid_credentials(login):
    login.with_("tomsmith", "SuperSecretPassword!")
    assert login.success_message_present()

def test_invalid_credentials(login):
    login.with_("tomsmith", "bad password")
    assert login.success_message_present() == False